contrib/dracut/90zfs/zfs-load-key.sh.in

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# This script only gets executed on systemd systems, see mount-zfs.sh for non-systemd systems
+
+# import the libs now that we know the pool imported
+[ -f /lib/dracut-lib.sh ] && dracutlib=/lib/dracut-lib.sh
+[ -f /usr/lib/dracut/modules.d/99base/dracut-lib.sh ] && dracutlib=/usr/lib/dracut/modules.d/99base/dracut-lib.sh
+. "$dracutlib"
+
+# load the kernel command line vars
+[ -z "$root" ] && root=$(getarg root=)
+# If root is not ZFS= or zfs: or rootfstype is not zfs then we are not supposed to handle it.
+[ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
+
+# There is a race between the zpool import and the pre-mount hooks, so we wait for a pool to be imported
+while true; do
+    zpool list -H | grep -q -v '^$' && break
+    [[ $(systemctl is-failed zfs-import-cache.service) == 'failed' ]] && exit 1
+    [[ $(systemctl is-failed zfs-import-scan.service) == 'failed' ]] && exit 1
+    sleep 0.1s
+done
+
+# run this after import as zfs-import-cache/scan service is confirmed good
+if [[ "${root}" = "zfs:AUTO" ]] ; then
+    root=$(zpool list -H -o bootfs)
```

Comment:
Does this work correctly with multiple pools imported?

Reply:
no, but the existing code didn't work in the first place either

```bash
# zpool list -H -o bootfs
deleteme
slaanesh-zp00/OS
```

https://github.com/zfsonlinux/zfs/blob/master/contrib/dracut/90zfs/zfs-lib.sh.in#L37 is where it's currently broken

I guess I can update it to do the same (just return the first dataset).

Reply:
We could return the first _likely useful_ dataset:

```
root@linux:~# zpool list -H -o name,bootfs
backup	-
rpool	rpool/boot
rpool2	rpool2/boot
root@linux:~# zpool list -H -o bootfs | head -n 1
-
root@linux:~# zpool list -H -o bootfs | awk '$1 != "-" {print; exit}'
rpool/boot
root@linux:~# 
```